### PR TITLE
Add `IrregularParameterGrid` class

### PR DESCRIPTION
**Pull request by tomaskontrimas**
It is needed when the binning of KDE PDFs is irregular in fit parameter.

### CODE REVIEW

File: skyllh/core/parameters.py [hunk 1]
```diff
@@ -1346,6 +1346,186 @@ def round_to_upper_grid_point(self, value):
...
+    @grid.setter
+    def grid(self, arr):
+        if not issequence(arr):
+            raise TypeError(
+                'The grid property must be a sequence!')
+        if not isinstance(arr, np.ndarray):
+            arr = np.array(arr, dtype=np.float64)
+        if arr.ndim != 1:
+            raise ValueError(
+                'The grid property must be a 1D numpy.ndarray!')
```
Comment (chiarabellenghi) [inline skyllh/core/parameters.py:1422]: Minor comment: couldn't it just accept any kind of 1D sequence and cat it to a numpy.ndarray?

File: skyllh/core/parameters.py [hunk 2]
```diff
@@ -1346,6 +1346,186 @@ def round_to_upper_grid_point(self, value):
...
+        self.grid = newgrid
+
+    def copy(self):
+        """Copies this IrregularParameterGrid object and returns the copy.
+        """
+        copy = deepcopy(self)
+        return copy
+
+    def round_to_nearest_grid_point(self, value):
+        """Rounds the given value to the nearest grid point.
```
Comment (chiarabellenghi) [inline skyllh/core/parameters.py:1450]: Maybe specify that the lower nearest point is chosen when the given value is precisely in between 2 grid points.

**Review by chiarabellenghi — APPROVED** [review]
I have two minor comments, but changes can be merged for me :)

File: skyllh/core/parameters.py (hunk 1 above)
Comment (tomaskontrimas) [reply skyllh/core/parameters.py:1422]: It does that already

File: skyllh/core/parameters.py (hunk 2 above)
Comment (tomaskontrimas) [reply skyllh/core/parameters.py:1450]: Done!

File: skyllh/core/parameters.py [hunk 3]
```diff
@@ -1364,7 +1547,7 @@ def __init__(
         """
         super().__init__(
             objs=param_grids,
-            obj_type=ParameterGrid,
+            obj_type=type(param_grids[0]),
```
Comment (martwo) [inline skyllh/core/parameters.py:1550]: This will fail, when the param_grids parameter is set to None!

Comment (martwo) [reply skyllh/core/parameters.py:1550]: @tomaskontrimas Ping!

**Pull request merged**